Programs/python.c

```diff
@@ -53,9 +60,37 @@ main(int argc, char **argv)
     fedisableexcept(FE_OVERFLOW);
 #endif
 
+    /* UTF-8 mode */
+    char *old_ctype = setlocale(LC_CTYPE, NULL);
+    if (old_ctype != NULL) {
+        old_ctype = _PyMem_RawStrdup(old_ctype);
```

Comment:
Maybe, it can be:

```c
    if (_Py_LegacyLocaleDetected()) {
        Py_UTF8Mode = 1;
        _Py_CoerceLegacyLocale();
    }
```

See here
https://github.com/python/cpython/blob/4ae06c5337e01bdde28bce57b6b9166ad50947e3/Programs/python.c#L75-L77

Reply:
"if (_Py_LegacyLocaleDetected()) { Py_UTF8Mode = 1;"

Right. It works as expected, I updated my PR.